### ISSUE TITLE
Incorrect heading use and structure on the patterns pages

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#3174]
## Description  
Headings should communicate the organization of the content on the page. If the heading levels are not logical, or if the heading is not used to communicate the organization of the content on the page,  screen reader users may be confused.

For example, on https://clarity.design/documentation/color, the first heading is a `<h5>`. Document outlines should begin with a `<h1>`.  Additionally, this heading does not contain a simple description of the content it represents. Instead it contains a list of content. This causes unusual behavior when using VoiceOver/Safari/iOS 12.1.4. The bullets in the list are announced as duplicate but empty headers.

See the relevant HTML below.

`<h5 _ngcontent-c3="" class="component-summary" id="the-clarity-color-palette-is-primarily-based-on-alignment-with-color-use-in-web-applications-and-support-for-accessibility"><ul _ngcontent-c3=""><li _ngcontent-c3=""><a _ngcontent-c3="" fragment="color-palette" routerlink="." href="/documentation/color#color-palette">Clarity Color Palette</a></li><li _ngcontent-c3=""><a _ngcontent-c3="" fragment="functional-colors" routerlink="." href="/documentation/color#functional-colors">Functional Colors</a></li><li _ngcontent-c3=""><a _ngcontent-c3="" fragment="chart" routerlink="." href="/documentation/color#chart">Chart Color Schemes</a></li></ul><p _ngcontent-c3="">The Clarity color palette is a full spectrum of bold hues and neutrals to meet all your design needs.</p></h5>`

Additionally, the patterns pages are missing `<h2>` headings and levels are being skipped.  Skipping heading levels can be confusing and should be avoided where possible: For example, make sure that a `<h1> `is not followed directly by an `<h5>`. It is ok to skip levels when closing subsections, for instance, a `<h2>` beginning a new section, can follow an` <h4>` as it closes the previous section, 

## Location    
Individual patterns selectable from the clarity components menu 

## Priority     
Medium

## Recommended fix       
Review the following tutorial 

https://www.w3.org/WAI/tutorials/page-structure/headings/

then fix

## LOE  
High, because there are a lot of them

**Comment by gnomeontherun** [comment]
Closing as duplicate of #3166
